lib/templateFactory.js

```diff
@@ -273,24 +287,33 @@ class TemplateFactory extends BaseFactory {
                 metricBuildsListConfig.endTime = endTime;
             }
 
-            return Promise.all([jobFactory.list(metricJobsListConfig), buildFactory.list(metricBuildsListConfig)]).then(
-                ([jCount, bCount]) =>
-                    templates.map(t => {
-                        const jobCount = jCount.find(j => j.templateId === t.id);
-
-                        const buildCount = bCount.find(b => b.templateId === t.id);
-
-                        t.metrics = {
-                            jobs: {
-                                count: jobCount ? jobCount.count : 0
-                            },
-                            builds: {
-                                count: buildCount ? buildCount.count : 0
-                            }
-                        };
-
-                        return t;
-                    })
+            return Promise.all([
+                jobFactory.list(metricJobsListConfig),
+                buildFactory.list(metricBuildsListConfig),
+                jobFactory.list(metricPipelinesConfig)
```

Comment:
Tested against SQLite DB and noticed below issues

1. The DB query results includes pipelines that do not have any job using template
2. It includes the pipelines with jobs using any template. Instead it should only include pipelines having jobs using the only the template specified in the config

**Query result:**
```
[
  {
    "pipelineId": 1,
    "count": 2
  },
  {
    "pipelineId": 2,
    "count": 2
  },
  {
    "pipelineId": 3,
    "count": 2
  },
  {
    "pipelineId": 4,
    "count": 3
  },
  {
    "pipelineId": 5,
    "count": 3
  },
  {
    "pipelineId": 6,
    "count": 3
  }
]
```

**Actual Jobs (Pipelines) using template:**
Only pipelines `4`, `5` and `6` have jobs using templates.
```
select pipelineId, id, name, templateId
from jobs
order by pipelineId;
```
```
+----------+--+-----------------------------+----------+
|pipelineId|id|name                         |templateId|
+----------+--+-----------------------------+----------+
|1         |1 |main                         |NULL      |
|1         |2 |publish                      |NULL      |
|2         |3 |validate                     |NULL      |
|2         |4 |publish                      |NULL      |
|3         |5 |validate                     |NULL      |
|3         |6 |publish                      |NULL      |
|4         |7 |job_not_using_template       |NULL      |
|4         |8 |job_using_template_apple_main|3         |
|4         |9 |job_using_template_apple_test|3         |
|5         |10|job_not_using_template       |NULL      |
|5         |11|job_using_template_pear_main |4         |
|5         |12|job_using_template_pear_test |4         |
|6         |13|job_not_using_template       |NULL      |
|6         |14|job_using_template_apple     |3         |
|6         |15|job_using_template_pear      |4         |
+----------+--+-----------------------------+----------+

```